# Test generated by RoostGPT for test test25OctPostmanCheck using AI Type Open AI and AI Model gpt-4

# RoostTestHash=2db2056e97

config:
  target: "{{ $processEnvironment.API_HOST }}"
  plugins:
    expect: {}
  payload:
    path: "pet_petId_get_others.csv"
    fields: ["petId"]
    skipHeader: true
  phases:
    - duration: 60
      arrivalRate: 20

scenarios:
  - name: "Testing failure cases for /pet/{petId} API"
    flow:
      - think: 1
      - log: "Sending request with an invalid petId"
        get:
          url: "/pet/invalidPetId"
          headers:
            Authorization: "{{ $processEnvironment.AUTH_TOKEN }}"
          expect:
            - statusCode: 400
            - hasProperty: 'description'
            - equals: 
                - "description": "Invalid ID supplied"
      - think: 1
      - log: "Sending request with a petId that does not exist"
        get:
          url: "/pet/{{ petId }}"
          headers:
            Authorization: "{{ $processEnvironment.AUTH_TOKEN }}"
          expect:
            - statusCode: 404
            - hasProperty: 'description'
            - equals: 
                - "description": "Pet not found"
      - think: 1
      - log: "Sending request with invalid authorization"
        get:
          url: "/pet/{{ petId }}"
          headers:
            Authorization: "InvalidToken"
          expect:
            - statusCode: 401
      - think: 1
      - log: "Sending request to an invalid URL"
        get:
          url: "/invalidUrl/{{ petId }}"
          headers:
            Authorization: "{{ $processEnvironment.AUTH_TOKEN }}"
          expect:
            - statusCode: 404
